add custom test hook

0atman opened the pull request:
In order to run standard, I've modified `npm test`, so it now that runs standard followed by jest.

Two new npm targets are added by this PR:

1. `npm run lint`
2. `npm run lint-fix`

- [x] Add standard and jest commands to test.sh
- [x] update package.json with standard
- [x] not expect es6 in JS we ship to users [fixed with lint exception headers]
- [x] be fine with ES6 in test files [fixed, as Standard :ok_hand:]
- [x] ignore config files

0atman commented:
had complications installing npm 7 yesterday, hence no package changes, gonna seek advice today!

kr8n3r commented:
is there a reason we cannot run standard as part of `test: jest` with say a config?
```
"standard": {
    "ignore": [
      "out/*"
    ],
    "env": [
      "jest",
      "es6",
      "browser"
    ],
```

0atman commented:
@kr8n3r I couldn't see how that's done https://standardjs.com/index.html#what-about-mocha-jest-jasmine-qunit-etc

0atman commented:
Standard config added to package.json

kr8n3r commented:
probably not directly in jest but I guess
```
 "lint": "standard",
 "test": "npm run lint && jest",
```
should work.

This gives an optional linting command for dev and the test should fail if linting fails, right?

0atman commented:
I like that, and the overhead of running 2 npm processes is worth the deduplication of `standard` commands :ok_hand:

kr8n3r commented:
now just need to figure out if we can configure standard (or use standardx) to 
- not expect es6 in JS we ship to users
- be fine with ES6 in test files 
- and ignore config files

do you want to do that in this PR?

0atman commented:
yeah, certainly, be nice to get add it all nicely, I'll get configing :ok_hand:

0atman commented:
Here's the remainder of the standard errors, I think they are now all just layout:

```
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:5:1: Expected indentation of 0 spaces but found 1.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:7:36: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:7:39: Block must not be padded by blank lines.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:17:3: Split initialized 'var' declarations into multiple statements.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:18:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:19:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:20:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:21:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:22:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:23:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:23:74: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:24:9: '&&' should be placed at the end of the line.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:32:1: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:43:3: Split initialized 'var' declarations into multiple statements.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:55:1: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:59:65: Block must not be padded by blank lines.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:61:3: Split initialized 'var' declarations into multiple statements.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:62:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:66:1: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:70:48: Block must not be padded by blank lines.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:72:3: Split initialized 'var' declarations into multiple statements.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:73:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:74:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:78:1: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.js:85:5: 'dataLayer' is not defined.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:1:8: 'React' is defined but never used.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:1:26: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:31:6: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:38:1: More than 1 blank line not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:39:6: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:39:104: Block must not be padded by blank lines.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:42:1: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:45:7: Unnecessarily quoted property '0' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:46:7: Unnecessarily quoted property '1' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:47:7: Unnecessarily quoted property '2' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:49:4: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:52:6: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:52:142: Block must not be padded by blank lines.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:58:7: Unnecessarily quoted property '0' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:59:7: Unnecessarily quoted property '1' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:60:7: Unnecessarily quoted property '2' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:65:135: Block must not be padded by blank lines.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:71:7: Unnecessarily quoted property '0' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:72:7: Unnecessarily quoted property '1' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:73:7: Unnecessarily quoted property '2' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:81:7: Unnecessarily quoted property '0' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:82:7: Unnecessarily quoted property '1' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:83:7: Unnecessarily quoted property '2' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/analytics.test.js:87:1: Too many blank lines at the end of file. Max of 0 allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:7:1: Expected indentation of 2 spaces but found 3.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:8:49: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:12:66: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:16:43: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:16:46: Block must not be padded by blank lines.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:19:45: Block must not be padded by blank lines.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:21:1: Expected indentation of 4 spaces but found 3.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:21:51: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:26:4: Block must not be padded by blank lines.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:27:1: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:42:81: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:44:77: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:47:91: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:49:63: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:50:48: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:51:18: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:54:92: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:56:63: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:57:49: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:58:18: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:75:52: Unnecessarily quoted property 'analytics' found.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:75:86: A space is required after '{'.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:75:112: A space is required before '}'.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:80:26: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:96:124: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:104:2: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:138:45: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:148:71: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:153:30: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:157:3: Split initialized 'var' declarations into multiple statements.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:158:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:166:29: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:172:3: Split initialized 'var' declarations into multiple statements.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:172:50: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:173:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:174:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:175:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:177:8: There should be no space after this paren.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:177:46: There should be no space before this paren.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:179:1: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:181:22: Unnecessary use of boolean literals in conditional expression.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:181:38: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:183:39: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:184:12: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:190:65: A space is required after ','.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:197:60: A space is required after '{'.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:197:86: A space is required before '}'.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:199:33: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:201:3: Expected space(s) after "if".
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:205:15: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:209:3: Split initialized 'var' declarations into multiple statements.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:210:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:211:1: Expected indentation of 4 spaces but found 6.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:216:37: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:217:38: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:219:38: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:222:39: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:232:96: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:232:96: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:233:44: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:233:45: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:233:45: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:234:16: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:234:18: Requires a space after '{'.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:234:18: Missing space before opening brace.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:234:19: 'dataLayer' is not defined.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:234:44: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:234:45: Requires a space before '}'.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:234:46: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:234:46: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:235:25: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:235:26: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:235:26: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:244:45: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:248:4: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:250:42: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:254:45: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:254:54: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:255:23: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:256:29: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:256:38: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:257:29: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/javascripts/cookie-functions.js:257:36: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/jest.config.js:4:3: Unnecessarily quoted property 'moduleNameMapper' found.
  /home/oatman/projects/gds/paas/paas-product-pages/jest.config.js:4:3: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/jest.config.js:5:5: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/jest.config.js:5:25: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/jest.config.js:6:5: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/jest.config.js:6:22: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/jest.config.js:8:2: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:3:44: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:4:7: 'babelPluginExtractFrontmatter' is assigned a value but never used.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:4:100: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:8:35: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:12:19: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:13:53: A space is required after ','.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:14:40: There should be no space after this paren.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:15:84: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:16:5: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/lib/rehype-removeTOC.js:20:1: Expected indentation of 8 spaces but found 10.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:3:21: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:3:46: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:4:27: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:4:52: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:13:9: A space is required after ','.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:14:18: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:14:24: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:14:30: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:14:36: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:16:14: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:17:15: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:18:19: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:19:15: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:19:30: Unexpected trailing comma.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:21:29: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:23:45: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:24:65: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:34:17: Unnecessarily quoted property 'type' found.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:34:17: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:34:25: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:35:17: Unnecessarily quoted property 'value' found.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:35:17: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:35:26: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:50:34: Unexpected trailing comma.
  /home/oatman/projects/gds/paas/paas-product-pages/next.config.js:51:4: Unexpected trailing comma.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_app.js:1:37: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:7:31: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:12:9: Missing space before function parentheses.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:15:40: Expected a line break after this opening brace.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:15:40: A space is required after '{'.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:42:10: Expected a line break before this closing brace.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:42:10: A space is required before '}'.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:42:13: The closing bracket must be aligned with the opening tag (expected column 9 on the next line)
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:43:25: Unexpected usage of doublequote.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:44:25: Unexpected usage of doublequote.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:44:70: A space is required after '{'.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:44:79: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:44:192: A space is required before '}'.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:44:197: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:45:19: Unexpected usage of doublequote.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:45:45: Unexpected usage of doublequote.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:50:24: Unexpected usage of doublequote.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:50:48: Unexpected usage of doublequote.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:50:72: Unexpected usage of doublequote.
  /home/oatman/projects/gds/paas/paas-product-pages/pages/_document.js:57:29: Newline required at end of file but not found.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:1:50: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:2:47: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:4:46: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:10:45: Extra semicolon.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:12:8: Expected space(s) after "default".
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:12:16: Trailing spaces not allowed.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:13:1: Expected indentation of 2 spaces but found 1 tab.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:13:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:14:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:14:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:15:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:15:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:16:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:16:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:17:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:17:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:18:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:18:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:19:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:19:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:20:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:20:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:21:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:21:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:22:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:22:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:23:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:23:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:24:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:24:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:25:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:25:1: Expected indentation of 2 spaces but found 1 tab.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:26:1: Expected indentation of 2 spaces but found 1 tab.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:26:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:27:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:27:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:27:10: Strings must use singlequote.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:28:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:28:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:29:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:29:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:29:53: Unexpected trailing comma.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:30:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:30:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:31:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:31:1: Expected indentation of 2 spaces but found 1 tab.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:32:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:32:1: Expected indentation of 2 spaces but found 1 tab.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:33:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:33:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:34:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:34:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:35:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:35:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:35:44: Unexpected trailing comma.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:36:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:36:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:37:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:37:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:38:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:38:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:39:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:39:1: Expected indentation of 8 spaces but found 4 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:40:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:40:1: Expected indentation of 8 spaces but found 4 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:41:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:41:1: Expected indentation of 8 spaces but found 4 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:42:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:42:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:43:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:43:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:44:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:44:1: Expected indentation of 2 spaces but found 1 tab.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:45:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:45:1: Expected indentation of 2 spaces but found 1 tab.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:46:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:46:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:47:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:47:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:48:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:48:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:49:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:49:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:50:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:50:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:51:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:51:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:52:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:52:1: Expected indentation of 8 spaces but found 4 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:53:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:53:1: Expected indentation of 8 spaces but found 4 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:54:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:54:1: Expected indentation of 8 spaces but found 4 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:55:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:55:1: Expected indentation of 10 spaces but found 5 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:56:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:56:1: Expected indentation of 10 spaces but found 5 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:57:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:57:1: Expected indentation of 10 spaces but found 5 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:58:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:58:1: Expected indentation of 10 spaces but found 5 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:59:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:59:1: Expected indentation of 10 spaces but found 5 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:60:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:60:1: Expected indentation of 8 spaces but found 4 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:61:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:61:1: Expected indentation of 6 spaces but found 3 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:62:1: Expected indentation of 4 spaces but found 2 tabs.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:62:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:63:1: Unexpected tab character.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:63:1: Expected indentation of 2 spaces but found 1 tab.
  /home/oatman/projects/gds/paas/paas-product-pages/rollup.config.js:64:2: Extra semicolon.
```

0atman commented:
@kr8n3r I'm going to make an atomic commit running `standard --fix` against the above layout errors, I think that'll now be safe? Just want to check, before I make the same mistake as last week!

0atman commented:
Also, @kr8n3r, it's easy to ignore `*config*` files, but should we? I mean, why allow errors in them?

kr8n3r commented:
I'm not going to push anything to this pr so go ahead as you wish. and yes a separate commit with --fix is good practice

0atman commented:
Perfection

kr8n3r commented:
> Also, @kr8n3r, it's easy to ignore `*config*` files, that's now committed, but should we? I mean, why allow errors in them?

true. fix them

0atman commented:
Standard fixes applied, the following lint errors remain:

- [ ] `analytics.js:82:5`: 'dataLayer' is not defined.
- [ ] `analytics.test.js:1:8`: 'React' is defined but never used.
- [ ] `cookie-functions.js:231:22`: 'dataLayer' is not defined.
- [ ] `lib/rehype-removeTOC.js:4:7`: 'babelPluginExtractFrontmatter' is assigned a value but never used.

kr8n3r commented on package.json:
```diff
@@ -9,7 +9,9 @@
     "start": "next start",
-    "test": "jest",
+    "test": "npm run lint && jest",
+    "lint": "standard",
```
looking ahead, if we do get around to implementing stylelint for scss files, can we use the same command or should we split them to lint:js lint:scss.
your call, I'm happy either way

0atman replied:
good question. maybe `lint` does it all, by calling `lint:js` etc, in the same way `test` does it all, but the developer can, if they choose, run them separately.

kr8n3r replied:
cool

kr8n3r commented:
> Standard fixes applied, the following lint errors remain:
> 
> * [ ]  `analytics.js:82:5`: 'dataLayer' is not defined.
> * [ ]  `analytics.test.js:1:8`: 'React' is defined but never used.
> * [ ]  `cookie-functions.js:231:22`: 'dataLayer' is not defined.
> * [ ]  `lib/rehype-removeTOC.js:4:7`: 'babelPluginExtractFrontmatter' is assigned a value but never used.

I'll have a look and double check lovely tmrw

0atman commented:
@kr8n3r nice, will fix both of those now in this PR

0atman commented:
```
 PASS  components/Button/button.test.tsx
 PASS  javascripts/analytics.test.js

Test Suites: 2 passed, 2 total
Tests:       9 passed, 9 total
Snapshots:   0 total
Time:        2.411 s
Ran all test suites.
```

ready to review, @kr8n3r  :ok_hand:

kr8n3r reviewed (approved):
:shipit: 
builds ok

0atman commented on package.json:
```diff
@@ -9,7 +9,9 @@
-    "test": "jest",
+    "test": "npm run lint && jest",
+    "lint": "standard",
+    "lint-fix": "standard --fix",
```
beautiful, done :ok_hand:
